python/data/format.py

```diff
@@ -17,7 +17,32 @@ def __post_init__(self):
         assert self.x.dtype == np.uint16
         assert self.y.dtype == np.uint16
         assert self.p.dtype == np.uint8
-        assert self.t.dtype == np.int64
+        assert self.t.dtype == np.uint32
```

Comment:
This is dangerous since we don't check that an overflow is not happening.

Reply:
Same for other loc that are affected by this

Reply:
Ok I am conviced, I have changed it back to int64. The memory footprint then grows from 12MB -> 13MB, which should be fine.